src/test/java/nl/tudelft/ewi/devhub/webtests/NotificationTest.java

```diff
@@ -0,0 +1,72 @@
+package nl.tudelft.ewi.devhub.webtests;
+
+import com.google.inject.Inject;
+import nl.tudelft.ewi.devhub.server.Config;
+import nl.tudelft.ewi.devhub.webtests.utils.WebTest;
+import nl.tudelft.ewi.devhub.webtests.views.LoginView;
+import nl.tudelft.ewi.devhub.webtests.views.NotificationView;
+import org.junit.Test;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+/**
+ * Created by Arjan on 14-6-2017.
+ */
+public class NotificationTest extends WebTest {
+
+    private static final String NET_ID = "admin1";
+    private static final String PASSWORD = "admin1";
+
+    @Inject
+    private Config config;
+
+    private static final By MARK_READ_BUTTON = By.xpath("/html/body/div/table/tbody/tr[2]/td/form/button");
+    private static final By NOTIFICATION_INDICATOR = By.xpath("//*[@id=\"bs-example-navbar-collapse-1\"]/ul/li[1]/a/span");
+    private static final By NOTIFICATIONS_READ = By.xpath("//*[@class=\"notification read\"]");
+    private static final By NOTIFICATIONS_UNREAD = By.xpath("//*[@class=\"notification unread\"]");
+
+
+    @Test
+    public void markAsRead() throws InterruptedException {
+
+        getDriver().navigate().to("http://localhost:"  + config.getHttpPort() + "/notifications/" + NET_ID);
+
+        //String url = getDriver().getCurrentUrl();
+        //notificationView.waitUntilCurrentUrlDiffersFrom(url);
+
+        LoginView loginView = new LoginView(getDriver());
+        loginView.setPasswordField(PASSWORD);
+        loginView.setUsernameField(NET_ID);
+
+        String url = getDriver().getCurrentUrl();
+
+        loginView.clickLoginButton();
+
+        loginView.waitUntilCurrentUrlDiffersFrom(url);
+
+        NotificationView notificationView  = new NotificationView(getDriver());
+        notificationView.invariant();
+
+        WebElement button = getDriver().findElement(MARK_READ_BUTTON);
```

Comment:
To be consistent with our Page Objects structure, the tests itself should not have to deal with XPath expressions or the webdriver. Instead this logic should be abstracted in the Page Objects itself. The `NotificationView` should expose an API such as: `notificationView().getUnreadNotifications().get(0).markRead()` and `notificationView().getUnreadNotifications().size()`.